src/Tribe/Aggregator/Record/Queue_Realtime.php

```diff
@@ -110,35 +111,48 @@ public function render_update_message() {
 	}
 
 	/**
-	 * Handle queue ajax requests
+     * Action to reply every time a heart beat is executed to send the progress of EA if an EA record is present.
+     *
+	 * @param array<string, mixed> $response The current response object
+	 * @param array<string, mixed> $data An array with the data from the client.
+	 *
+	 * @return array<string, mixed> An array used to construct the heart beat response.
 	 */
-	public function ajax() {
-		$this->record_id = (int) $_POST['record'];
+	public function receive_heartbeat( array $response, array $data ) {
+		if ( empty( $data['ea_record'] ) ) {
+			return $response;
+		}
 
-		// Nonce check
-		$this->ajax_operations->verify_or_exit( $_POST['check'], $this->get_ajax_nonce_action(), $this->get_unable_to_continue_processing_data() );
+		$this->record_id = absint( $data['ea_record'] );
 
-		// Load the queue
-		/** @var \Tribe__Events__Aggregator__Record__Queue_Interface $queue */
-		$queue = $this->queue ? $this->queue : Tribe__Events__Aggregator__Record__Queue_Processor::build_queue( $this->record_id );
-		// We always need to setup the Current Queue
-		$this->queue_processor->set_current_queue( $queue );
+		if ( $this->record_id === 0 ) {
+			return $response;
+		}
 
-		// Only if it's not empty that we care about processing.
-		if ( ! $queue->is_empty() ) {
-			$this->queue_processor->process_batch( $this->record_id );
+		$data = $this->get_queue_progress_data();
+
+		if ( empty( $data ) ) {
+			return $response;
 		}
 
-		/**
-		 * Include current queue to prevent progress bar from sticking on csv imports
-		 *
-		 * @var \Tribe__Events__Aggregator__Record__Queue_Interface $current_queue
-		 */
-		$current_queue = $this->queue_processor->current_queue;
-		$done          = $current_queue->is_empty() && empty( $current_queue->is_fetching() );
+		$response['ea_progress'] = $data;
 
-		$percentage = $current_queue->progress_percentage();
-		$this->ajax_operations->exit_data( $this->get_progress_message_data( $current_queue, $percentage, $done ) );
+		return $response;
+	}
+
+	/**
+	 * Handle queue ajax requests
+	 */
+	public function ajax() {
+	    _deprecated_function( __METHOD__, 'TBD' );
```

Comment:
:no_entry_sign: Tabs must be used to indent lines; spaces are not allowed

<sub>hash: 70c3bd3e8cee83eea7219fa813a0f6fb</sub>